client/state/index.js

```diff
@@ -80,6 +81,7 @@ const reducers = {
 	currentUser,
 	documentHead,
 	domains,
+	dropZone,
```

Comment:
I'd avoid adding a top-level branch to state for something as narrow as DropZone, and I'd avoid the precedent of having stuff in `state/components`.

Rationale:

- React components are not singletons and thus it doesn't make sense for a component to have its place in global state. If it's really the component (i.e., the implementation) that needs its own state, it can either live as a static property of the component or as instance state via `setState`.
- The real semantics we seek aren't that the _component_ needs a branch of state, but rather that _our application_ has a stateful UI and that we'd like UI state be aware of dropzones. Thus, my reflex would be to add `dropzone` to `state.ui`. If you look at [`state/ui/reducer`](https://github.com/Automattic/wp-calypso/blob/ca140478ee853279a566739bd044895e60a043bb/client/state/ui/reducer.js), you'll see some of the sub-reducers are defined in the very file and some are modules in `state/ui/` — your choice. :)